Xamarin.PropertyEditing.Tests/CombinablePredefinedViewModelTests.cs

```diff
@@ -0,0 +1,383 @@
+			Assert.That (flag3Choice.IsFlagged.HasValue && flag3Choice.IsFlagged.Value, Is.True);
+			Assert.That (vm.Choices.Count (v => v.IsFlagged.HasValue && !v.IsFlagged.Value), Is.EqualTo (5), "Unselected choices not marked false");
+		}
+
+		[Test]
+		public void GetsFlagsDisasgree ()
+		{
+			FlagsTestEnum value = FlagsTestEnum.Flag2 | FlagsTestEnum.Flag3;
+			FlagsTestEnum value2 = FlagsTestEnum.Flag2 | FlagsTestEnum.Flag4;
+
+			var p = GetPropertyMock ();
+
+			var target = new object();
+			var editorMock = new Mock<IObjectEditor> ();
+			editorMock.SetupGet (e => e.Target).Returns (target);
+			editorMock.Setup (e => e.GetValueAsync<IReadOnlyList<int>> (p.Object, null)).ReturnsAsync (
+				new ValueInfo<IReadOnlyList<int>> {
+					Value = new int[] { (int) FlagsTestEnum.Flag2, (int) FlagsTestEnum.Flag3 },
+					Source = ValueSource.Local
+				});
+			editorMock.Setup (e => e.GetValueAsync<int> (p.Object, null)).ReturnsAsync (new ValueInfo<int> {
+				Value = (int)value,
+				Source = ValueSource.Local
+			});
+
+			var target2 = new object();
+			var editorMock2 = new Mock<IObjectEditor> ();
+			editorMock2.SetupGet (e => e.Target).Returns (target2);
+			editorMock2.Setup (e => e.GetValueAsync<IReadOnlyList<int>> (p.Object, null)).ReturnsAsync (
+				new ValueInfo<IReadOnlyList<int>> {
+					Value = new int[] { (int) FlagsTestEnum.Flag2, (int) FlagsTestEnum.Flag4 },
+					Source = ValueSource.Local
+				});
+			editorMock2.Setup (e => e.GetValueAsync<int> (p.Object, null)).ReturnsAsync (new ValueInfo<int> {
+				Value = (int)value2,
+				Source = ValueSource.Local
+			});
+
+			var vm = GetViewModel (p.Object, new[] { editorMock.Object, editorMock2.Object });
+			Assume.That (vm.Choices.Count, Is.EqualTo (7));
+
+			var flag2Choice = vm.Choices.Single (v => v.Name == nameof(FlagsTestEnum.Flag2));
+			Assume.That (flag2Choice.Value, Is.EqualTo ((int)FlagsTestEnum.Flag2));
+			var flag3Choice = vm.Choices.Single (v => v.Name == nameof(FlagsTestEnum.Flag3));
+			Assume.That (flag3Choice.Value, Is.EqualTo ((int)FlagsTestEnum.Flag3));
+			var flag4Choice = vm.Choices.Single (v => v.Name == nameof(FlagsTestEnum.Flag4));
+			Assume.That (flag4Choice.Value, Is.EqualTo ((int)FlagsTestEnum.Flag4));
+			var flag5Choice = vm.Choices.Single (v => v.Name == nameof(FlagsTestEnum.Flag5));
+			Assume.That (flag5Choice.Value, Is.EqualTo ((int)FlagsTestEnum.Flag5));
+
+			Assert.That (vm.Value, Is.EqualTo (default(int)));
+			Assert.That (vm.MultipleValues, Is.True, "MultipleValues is not marked true for disagreeing values");
+			Assert.That (flag2Choice.IsFlagged.HasValue && flag2Choice.IsFlagged.Value, Is.True);
+			Assert.That (flag3Choice.IsFlagged.HasValue, Is.False, "Disagreed upon choice is not indeterminate");
+			Assert.That (flag4Choice.IsFlagged.HasValue, Is.False, "Disagreed upon choice is not indeterminate");
+			Assert.That (flag5Choice.IsFlagged.HasValue, Is.False, "Disagreed upon duplicate choice is not indeterminate");
```

Comment:
When you have the time, I'd add to the comment, as this is confusing: since 5 has the same value as 4, you'd expect the behavior to be the same, except that one is explicitly included in the actual value during setup, and the other is sort of an accident. If I got that correctly...